doc/api/util.md

```diff
@@ -3655,6 +3655,89 @@ Returns `true` if the value is a built-in {WeakSet} instance.
 util.types.isWeakSet(new WeakSet());  // Returns true
 ```
 
+## Disposer APIs
+
+> Stability: 1 - Experimental
+
+### `util.asyncDisposer(onDispose)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `onDispose` {Function} A dispose function returning a promise
+* Returns: {AsyncDisposer}
+
+Create a convenient wrapper on the given async function that can be used
+with `using` declaration.
+
+If an error is thrown in the function, instead of returning a promise,
+the error will be wrapped in a rejected promise.
```

Comment:
Given that `DisposableStack.defer(...)` exists I'm not super convinced that this is needed or all that useful.

```js
{
  using ds = new DisposableStack();
  ds.defer(onDispose);
}
```